mlir/lib/Dialect/SparseTensor/Transforms/SparseTensorRewriting.cpp

```diff
@@ -299,8 +299,7 @@ struct FuseSparseMultiplyOverAdd : public OpRewritePattern<GenericOp> {
     Block &prodBlock = prod.getRegion().front();
     Block &consBlock = op.getRegion().front();
     IRMapping mapper;
-    Block *fusedBlock = new Block();
-    fusedOp.getRegion().push_back(fusedBlock);
+    Block *fusedBlock = rewriter.createBlock(&fusedOp.getRegion());
     unsigned num = prodBlock.getNumArguments();
     for (unsigned i = 0; i < num - 1; i++)
       addArg(mapper, fusedBlock, prodBlock.getArgument(i));
```

Comment:
I suspect this may be also unhappy if the underlying implementation adds arguments to fusedBlock.

Reply:
We have no `notifyBlockModified` callback at the moment, so I think there is no expectation that a listener be notified about adding/removing/changing block arguments. I have plans to add `RewriterBase::Listener::notifyBlockModified` and generalize `modifyOpInPlace` to `modifyInPlace` (accepting both ops and blocks).